### PR TITLE
Fix fenrir issues

### PR DESCRIPTION
**Pull request by embhorn**
## Fixed findings

  ### Medium
  - f-7172: WebSocket Origin allowlist fails open when the Origin header exceeds the 256-byte scratch buffer

  ### Low
  - f-7173: Broker sends PUBACK/PUBREC SUCCESS for a v5 PUBLISH it silently discards (unsupported Topic Alias)
  - f-7176: WebSocket outbound staging buffer `ws->tx_pending` freed without zeroization
  - f-7177: Broker retains plaintext CONNECT username for the whole session while the password is scrubbed
  - f-7178: MQTT v5 AUTH received through MqttClient_WaitMessage leaves AUTH_DATA in client rx_buf
  - f-7158: MqttSNClient_PacketReset omits the MqttPendResp offset for SN_MSG_TYPE_DISCONNECT
  - f-7159: MqttClient_PacketReset applies the MqttPendResp offset to ack-only types that have no pendResp member
  - f-7161: SN_WillTopic waits with a different packet_obj than the one registered in its pending-response entry
  - f-7174: Decoded MQTT v5 property list leaked on late error returns in MqttDecode_Publish and MqttDecode_SubscribeAck
  - f-7160: Unreachable VBI-overflow check in MqttPacket_Read after the in-loop  4-byte guard

  ### Info
  - f-7162: BrokerHandle_Subscribe reports the requested QoS as granted when the filter-length decode is skipped
  - f-7163: Redundant MQTT_CODE_CONTINUE test in MqttBroker_Run's else-if
  - f-7175: Dangling else spanning #ifdef boundaries in BrokerClient_Free

### CODE REVIEW

**Review by aidangarske — COMMENTED** [review]
## Skoll Code Review

**Scan type:** `review-security`
**Overall recommendation:** `COMMENT`
**Findings:** 2 total — 2 posted, 0 skipped
**2 finding(s) not tied to a diff line** (full detail below)

### Findings not tied to a diff line

#### UNSUBACK properties are freed inside the API and left dangling
**File:** `src/mqtt_client.c:2697-2823`
**Function:** `MqttClient_Unsubscribe`
**Severity:** Medium
**Category:** memory

MqttClient_Unsubscribe frees unsubscribe->ack.props before returning but does not clear the field. The response object is caller-provided and remains externally visible after the API returns, so callers can observe or reuse a stale pointer, causing use-after-free or double-free behavior.

**Recommendation:** After freeing unsubscribe->ack.props, set it to NULL, or leave ownership with the caller and document/free it consistently at the API boundary.

_Referenced code: `src/mqtt_client.c:2697-2700` (4 lines)_

---

#### Signed-length truncation bypasses the new 64 KiB string guard
**File:** `src/mqtt_packet.c:691-714`
**Function:** `MqttEncode_String`
**Severity:** Medium
**Category:** length-arithmetic

MqttEncode_String casts XSTRLEN(str) to int before validating the MQTT 65535-byte string limit. Very large strings can truncate during the cast, allowing malformed length encoding and an invalid copy length to reach XMEMCPY.

**Recommendation:** Keep the XSTRLEN result in size_t, reject values greater than 0xFFFF before narrowing, then cast only after validation.

_Referenced code: `src/mqtt_packet.c:691-697` (7 lines)_

---

*Review generated by [Skoll](https://github.com/wolfSSL/fenrir/tree/main/skoll)*

**Review by aidangarske — COMMENTED** [review]
## Skoll Multi-Scan Review

**Modes:** `review + review-security`
**Overall recommendation:** `COMMENT`
**Findings:** 3 total — 3 posted, 0 skipped
**3 finding(s) posted as inline comments** (see file-level comments below)

### Posted findings
- [Medium] [review] **Missing WebSocket regression coverage** — `src/mqtt_broker.c:916-1069`
- [Medium] [review] **Missing AUTH_DATA receive-path test** — `src/mqtt_client.c:1113-1120`
- [Medium] [review] **Missing property-cleanup error-path tests** — `src/mqtt_packet.c:2310-2316, 2937-2952`

*Review generated by [Skoll](https://github.com/wolfSSL/fenrir/tree/main/skoll)*

File: src/mqtt_broker.c [hunk 1]
```diff
@@ -913,6 +913,22 @@ static void BrokerClient_Remove(MqttBroker* broker, BrokerClient* bc);
 static void BrokerClient_PublishWill(MqttBroker* broker, BrokerClient* bc);
 #endif
 
+/* Scrub and release the WebSocket outbound staging buffer. The staged copy
```
Comment (aidangarske) [inline src/mqtt_broker.c:916]: **🟠 [Medium] Missing WebSocket regression coverage**

The new Origin-length handling and WebSocket staging-buffer cleanup have no focused regression tests. Boundary cases such as an exact-size or oversized Origin, mismatches, and cleanup after successful writes, short writes, and disconnects are therefore unverified.

**Fix:** Add focused WebSocket regression tests before merging.

File: src/mqtt_client.c [hunk 2]
```diff
@@ -1110,6 +1110,13 @@ static int MqttClient_HandlePacket(MqttClient* client,
             rc = MqttClient_DecodePacket(client, client->rx_buf,
                 client->packet.buf_len, packet_obj, &packet_type, &packet_qos,
                 &packet_id, 1);
+            /* The decoded AUTH properties (including AUTH_DATA) were pointers
```
Comment (aidangarske) [inline src/mqtt_client.c:1113]: **🟠 [Medium] Missing AUTH_DATA receive-path test**

No test was found that delivers an inbound MQTT v5 AUTH packet through MqttClient_WaitMessage and verifies that AUTH_DATA is scrubbed from rx_buf after callback processing and property cleanup.

**Fix:** Add a regression test for the WaitMessage AUTH cleanup path.

File: src/mqtt_packet.c [hunk 3]
```diff
@@ -2305,6 +2307,13 @@ int MqttDecode_Publish(byte *rx_buf, int rx_buf_len, MqttPublish *publish)
 
     /* Decode Payload */
     if (variable_len > remain_len) {
+    #ifdef WOLFMQTT_V5
```
Comment (aidangarske) [inline src/mqtt_packet.c:2310]: **🟠 [Medium] Missing property-cleanup error-path tests**

Existing malformed-packet coverage does not assert that v5 property lists are released and nulled on the newly covered late-error paths after properties have already been decoded.

**Fix:** Add targeted leak and pointer-state regression tests for these decoder error paths.

---
> **Note:** Referenced line (`src/mqtt_packet.c:2310-2316, 2937-2952`) is outside the diff hunk. Comment anchored to nearest changed region.

File: src/mqtt_broker.c (hunk 1 above)
Comment (embhorn) [reply src/mqtt_broker.c:916]: Not a useful test

File: src/mqtt_client.c (hunk 2 above)
Comment (embhorn) [reply src/mqtt_client.c:1113]: Not a useful test

**Review by aidangarske — APPROVED** [review]
